### PR TITLE
Fix most unit tests (only 3 errors remaining).

### PR DESCRIPTION
**Pull request by ValMenn**

### CODE REVIEW

File: src/Framework/UnitTests/Compat/TempFileCollection.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+//
+// Minimal implementation of TempFileCollection
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace Microsoft.Build.Framework.UnitTests
+{
+    class TempFileCollection : IDisposable
```
Comment (rainersigwald) [inline src/Framework/UnitTests/Compat/TempFileCollection.cs]: Where is this used?  How were tests building on full-framework?

**Comment by rainersigwald** [comment]
Can you describe what "fix" means in this context please? (Compiles for what framework?  Tests pass?).

**Comment by ValMenn** [comment]
Updated this PR: a compat implementation of TempFileCollection.cs was added to a wrong project originally - moved to the correct one.

Fix - means get it to compile. I did not run it. Specifically, it now compiles in *-NetCore configuration with only 3 errors in one file. All other build problems were fixed using compile-time defines (FEATURE_*). Majority were replaced with a working alternative. Some tests were completely removed from compilation if appropriate features were not available (app domain and config files, for example). In one case a minimum-functionality replacement class was added for the one missing (TempFileCollection). In order to do the work I used a private version of packages.config, which installed nunit. That change is not in the PR as we are getting ready for xunit.

**Comment by rainersigwald** [comment]
LGTM

**Pull request merged**